src/components/sbb-alert/sbb-alert.tsx

```diff
@@ -0,0 +1,238 @@
+
+  /** Emits when the fade in animation ends and the button is displayed. */
+  @Event({
+    eventName: 'sbb-alert_did-present',
+  })
+  public didPresent: EventEmitter<void>;
+
+  /** Emits when dismissal of an alert was requested. */
+  @Event({
+    eventName: 'sbb-alert_dismissal-requested',
+  })
+  public dismissalRequested: EventEmitter<void>;
+
+  private _transitionWrapperElement!: HTMLElement;
+  private _alertElement!: HTMLElement;
+
+  private _currentLangauge = getDocumentLang();
+  private _firstRenderingDone = false;
+
+  public componentDidRender(): void {
+    if (!this._firstRenderingDone) {
+      this._present();
+    }
+    this._firstRenderingDone = true;
+  }
+
+  /** Requests dismissal of the alert. */
+  @Method() public async requestDismissal(): Promise<void> {
+    this.dismissalRequested.emit();
+  }
+
+  /** Present the alert. */
+  private _present(): Promise<void> {
+    this.willPresent.emit();
+
+    if (this.disableAnimation) {
+      this._onHeightTransitionEnd();
+      return;
+    }
+
+    this._transitionWrapperElement.style.height = `${this._alertElement.offsetHeight}px`;
+    this._transitionWrapperElement.addEventListener(
+      'transitionend',
+      this._onHeightTransitionEnd.bind(this),
+      {
+        once: true,
+      }
+    );
+  }
+
+  private _initFadeInTransitionStyles(): void {
+    if (this.disableAnimation) {
+      return;
+    }
+    this._transitionWrapperElement.style.height = '0';
+    this._alertElement.style.opacity = '0';
+  }
+
+  private _onHeightTransitionEnd(): void {
+    this._transitionWrapperElement.style.removeProperty('height');
+    this._alertElement.style.removeProperty('opacity');
+
+    if (this.disableAnimation) {
+      this._onOpacityTransitionEnd();
+      return;
+    }
+
+    this._alertElement.addEventListener('transitionend', this._onOpacityTransitionEnd.bind(this), {
+      once: true,
+    });
+  }
+
+  private _onOpacityTransitionEnd(): void {
+    this.didPresent.emit();
+  }
+
+  private _linkProperties() {
+    return {
+      ['accessibility-label']: this.accessibilityLabel,
+      ['accessibility-describedby']: this.accessibilityDescribedby,
+      ['accessibility-labelledby']: this.accessibilityLabelledby,
+      href: this.href,
+      rel: this.rel,
+      target: this.target,
+    };
+  }
+
+  public render(): JSX.Element {
+    return (
+      <div
+        class="sbb-alert__transition-wrapper"
+        ref={(el): void => {
+          this._transitionWrapperElement = el;
+        }}
+      >
+        <div
+          class="sbb-alert"
+          ref={(el): void => {
+            const isFirstInitialization = !this._alertElement;
+
+            this._alertElement = el;
+            if (isFirstInitialization) {
+              this._initFadeInTransitionStyles();
+            }
+          }}
+        >
+          <span class="sbb-alert__icon">
+            <slot name="icon">{this.iconName && <sbb-icon name={this.iconName} />}</slot>
```

Comment:
Should it maybe default to info, if no iconName is set?

Reply:
maybe you meant we want to guarantee always an icon

Reply:
changed it